core/src/test/java/org/elasticsearch/discovery/DiscoveryWithServiceDisruptionsIT.java

```diff
@@ -961,6 +963,15 @@ public void testIndicesDeleted() throws Exception {
         assertFalse(client().admin().indices().prepareExists("test").get().isExists());
     }
 
+    @After
+    // make sure we do not leak pending cluster states
+    public void assertNoPendingClusterStates() throws Exception {
```

Comment:
Out of curiosity, do we want the check on a global level as well as just for this IT?